content/blog/day-2-kubernetes-migrating-eks-nodegroups-with-zero-downtime/index.md

```diff
@@ -0,0 +1,170 @@
+`v1.13` of Kubernetes, along with the cluster's infrastructure dependencies
+(VPC and IAM) using [Crosswalk for AWS][crosswalk-aws]. We'll also
+create and attach the following node groups to the cluster:
+
+* A standard `t2.medium` worker node group using the recent `v1.13.7` worker [AMI][eks-amis]. For use by general purpose workloads such as the [`EchoServer`][echoserver], a simple app that echo's client request headers.
+* A 2xlarge `t3.2xlarge` worker node group using an older `v1.12.7` worker [AMI][eks-amis]. For use use by larger, intensive workloads such as the [NGINX Ingress Controller][ingress-nginx].
+
+With the cluster successfully created and the node groups available, Pulumi
+will deploy the workload: the [`EchoServer`][echoserver], and the [NGINX Ingress Controller][ingress-nginx] that will manage it's ingress.
+The `echoserver` will land on the Standard node group, and NGINX is set
+to specifically target the `2xlarge` node group.
+
+Once the workload is deployed, we can validate it is up and running by accessing
+the `echoserver` behind the NGINX endpoint using `curl`:
+
+<img style="max-width:none; height: 350px; width: 850px;" src="k8s-cluster-workload.gif"/>
+
+> **Note**: You can open the <a href="./k8s-cluster-workload.gif" target="_blank">GIF</a> for a maximized view.
+
+## The Great Migration
+
+After initial deployment, we decide to update the node group used by NGINX.
+
+We'll move NGINX from the `2xlarge` node group over to a new, `4xlarge` worker
+node group that differs in: an updated AMI, instance type, and desired instance count.
+
+As we migrate NGINX over to the `4xlarge` and decommission the `2xlarge` node
+group in the next steps, we'll also actively load test the endpoint of the
+`echoserver` to ensure that we are not losing requests through out the migration.
+
+### Step 1: Create the new `4xlarge` Node Group
+
+Create the new, `4xlarge` node group by defining it and running an update.
+
+<p align="center"><img src="ng-4xlarge.svg" width="650"/></p>
+
+### Step 2: Migrate NGINX to the `4xlarge` Node Group
+
+With the `4xlarge` node group created, we'll migrate the NGINX service away
+from the `2xlarge` node group over to the `4xlarge` node group by changing
+its node selector scheduling terms.
+
+This change updates the NGINX Deployment spec to require the use of
+`c5.4xlarge` nodes during scheduling, and forces a rolling update over to the
+`4xlarge` node group.
+
+NGINX is able to successfully migrate across node groups because it is
+configured with HA settings, spread-type scheduling predicates,
+and can gracefully terminate within the Kubernetes [Pod lifecycle][pod-lifecycle].
+
+<p align="center"><img src="target-ng-4xlarge.svg" width="650"/></p>
+
+### Step 3: Decomission the `2xlarge` Node Group
+
+Once migration of NGINX has completed onto the `4xlarge` node group, we can
+commence the decommissioning of the original `2xlarge` node group no
```

Comment:
```suggestion
begin decommissioning the original `2xlarge` node group which is no longer in use.
```

I would just end with that instead of adding "from Kubernetes and AWS"